File: 화연/Lv.2/유사칸토어비트열.js
/**
 * https://school.programmers.co.kr/learn/courses/30/lessons/148652
 * @author HwaYeon
 * @param {*} n 
 * @param {*} l 
 * @param {*} r 
 * @returns 
 */
function solution(n, l, r) {
  let num = r - l + 1;
  for(let i = l; i <= r; i++) {
      let snum = i;
      if(snum === 3) {
          num--;
      }
      while(snum > 5) {
          if(snum%5 === 3) {
              num--;
              break;
          }
          snum = Math.ceil(snum/5); 
          if(snum === 3) {
              num--;
              break;
          }
      }  
  }
  return num;
}


/**
 * 천잰가??? 어떻게 5진수로 바꿔서 풀 생각을 함??
 * 마치 고3 때의 나같군 푸항항항항
 * @author 최종택 등
 * @param {*} n 
 * @param {*} l 
 * @param {*} r 
 * @returns 
 */
function solution(n, l, r) {
  let answer = 0;
  for (let i = l - 1; i <=r - 1; i++) {
    if (!i.toString(5).match('2')) answer += 1;
  }
  return answer;
}
  
  

File: 성재/Lv.2/유사_칸토어_비트열.java
/*
문제 링크 : https://school.programmers.co.kr/learn/courses/30/lessons/148652
 */

import java.util.*;

class Solution {
    public int solution(int[] priorities, int location) {
        // 총 bit 개수 (5의 n승)
        long max_bit_count = (long) Math.pow(5, n);
        int answer = 0;

        for (long idx = l; idx <= r; idx++) {
            long check_bit = idx;
            long bit_mod = max_bit_count / 5;
            boolean is_zero = false;

            while (bit_mod != 0) {
                // 해당 bit 위치가 0이 반복되는 위치에 존재할 경우 (11011 11011 '00000' 11011 11011)
                if ((check_bit > bit_mod * 2) && (check_bit <= bit_mod * 3)) {
                    is_zero = true;
                    break;
                }

                // 해당 bit 위치가 0이 반복되는 위치 이후에 존재할 경우 (11011 11011 00000 '11011 11011')
                if (check_bit > bit_mod * 3) {
                    // bit 위치 조정 (ex. 1번째 bit = 16번째 bit)
                    check_bit -= bit_mod * 3;
                }

                // 해당 bit 위치가 중복이 시작되는 위치에 존재할 경우 (11011 '11011' 00000 11011 11011)
                if (check_bit > bit_mod) {
                    // bit 위치 조정 (ex. 1번째 bit = 6번째 bit)
                    check_bit -= bit_mod;
                }

                bit_mod /= 5;
            }

            if (is_zero) {
                continue;
            }

            answer++;
        }

        return answer;
    }
}